recipes/platform.hashing/all/test_package/conanfile.py

```diff
@@ -1,28 +1,39 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanException
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake, CMakeToolchain, CMakeDeps
+from conan.tools.microsoft import is_msvc
 import os
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    test_type = "explicit"
 
     @property
     def _extra_flags(self):
-        return self.deps_user_info["platform.hashing"].suggested_flags
+        return self.dependencies["platform.hashing"].conf_info.get("user.platform_hashing:suggested_flags", check_type=str)
 
-    def build(self):
-        if self.settings.compiler != "Visual Studio":
-            if not self._extra_flags:
-                raise ConanException("Suggested flags are not available for os={}/arch={}".format(self.settings.os, self.settings.arch))
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
 
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if not is_msvc(self):
+            tc.variables["EXTRA_FLAGS"] = self._extra_flags
+            print(f"aaaaaa {self._extra_flags}")
```

Comment:
Is this some debug output here?

Reply:
Oh, it's my mistake!
I will fix soon.
Thanks!

Reply:
No problem, LGTM!